enhancements/upgrade-readiness.md

```diff
@@ -0,0 +1,291 @@
+kind: ClusterServiceVersion
+metadata:
+  name: foo-operator.v1.0.0
+  namespace: olm
+spec:
+  description: |
+    An example opreator
+  displayName: foo-operator
+  install:
+    spec:
+      deployments:
+      - name: foo-operator
+        upgradeReadinessServices:              # An array of readiness services
+        - name: foo-operator-readiness-service # The Service name
+          spec:                                # Standard Service spec
+            selector:                            
+              app: foo-operator
+            ports:
+            - protocol: TCP
+              port: 80
+              targetPort: 9376
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+                app: foo-operator
+          template:
+            metadata:
+              labels:
+                app: foo-operator
+            spec:
+              containers:
+              - name: foo-operator
+                image: foo
+                command:
+                  - sleep
+                  - "3600"
+    strategy: deployment
+  installModes:
+  - supported: false
+    type: OwnNamespace
+  - supported: true
+    type: SingleNamespace
+  - supported: true
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
+  maturity: alpha
+  provider:
+    name: Red Hat
+  version: 1.0.0
+```
+##### Approach 2: Omitting the Upgradeable Readiness Service Spec
+
+This approach allows operator authors to provide the name of a upgradeable readiness service that should appear on cluster. When using this approach, the service will typically be present in the operator bundle.
+
+First, you must create a CSV that omits the spec of the upgradeReadinessService:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: foo-operator.v1.0.0
+  namespace: olm
+spec:
+  description: |
+    An example opreator
+  displayName: foo-operator
+  install:
+    spec:
+      deployments:
+      - name: foo-operator
+        readinessServices:                     # An array of upgradeReadiness services
+        - name: foo-operator-readiness-service # The name of the Service
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+                app: foo-operator
+          template:
+            metadata:
+              labels:
+                app: foo-operator
+            spec:
+              containers:
+              - name: foo-operator
+                image: foo
+                command:
+                  - sleep
+                  - "3600"
+    strategy: deployment
+  installModes:
+  - supported: false
+    type: OwnNamespace
+  - supported: true
+    type: SingleNamespace
+  - supported: true
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
+  maturity: alpha
+  provider:
+    name: Red Hat
+  version: 1.0.0
+```
+
+The operator bundle should include the following Service:
```

Comment:
Should there be an opm/sdk story for failing validation if no service object is found matching?